optaplanner-core/src/main/java/org/optaplanner/core/impl/score/director/stream/KieBaseDescriptor.java

```diff
@@ -8,7 +8,7 @@
 import org.optaplanner.core.impl.score.inliner.WeightedScoreImpacter;
 import org.optaplanner.core.impl.score.stream.drools.DroolsConstraint;
 
-final class KieBaseDescriptor<Solution_> {
+final public class KieBaseDescriptor<Solution_> {
```

Comment:
I am not prepared to make this a public API. Furthermore, the way you share this type with DRL now makes it kinda ugly - `constraintToGlobalMap` has no meaning there. I think the two wrappers will have to stay separate.

Reply:
The wrappers are separate (one uses `Supplier<KieBase>`, another use `Supplier<KieBaseDescriptor>`. I can try making it `Supplier<Object>` and doing typecasting.

Reply:
`KieBaseDescriptor` exposes things that I do not want users to access. The globals are an implementation detail which I absolutely do not want to leak into a public API.

I'm thinking we create some sort of internal config which subtype which is not public and allows us to pass this sort of things. But right now, I'm not sure that's the best thing to do. Will have to think about it a bit more.